### ISSUE TITLE
Build a Directus Powered WebNote Keeper Chrome Extension

### ISSUE THREAD

**Issue by Theodore-Kelechukwu-Onyejiaku** [directus-labs/guest-authoring#102]
### What is your idea?

The idea is to create a Chrome extension called "WebNote Keeper" that allows users to save notes related to webpages they visit. The extension will leverage Directus as the backend CMS to store and manage these notes. When a user clicks the extension while browsing a webpage, it will automatically capture the URL of the current webpage and prompt the user to add a note. Users will be able to view, edit, and delete their notes directly from the extension's interface. The extension will integrate with Directus API for CRUD operations on notes. This project combines the use of Directus as a headless CMS and Chrome extension development skills to create a practical tool for users to organize their web research and thoughts.

### What are the key takeaways from your post?

* Directus Integration
* Chrome Extension Development
* Directus API Integration
* User Interface Design
* Transferrable Skills
* Practical Application

### What type of post is this?

Full Project Build with Directus

### Country of residence

Nigeria

### Terms & Conditions

- [X] You understand that submitting this form does not mean we accept your proposal. We will accept proposals at our discretion.
- [X] You understand that to be paid you will need to generate an invoice to us once the final draft of your work is completed.
- [X] You understand that you will grant us an unlimited license to post and promote your work as part of this program. You may repost, but using our copy as the canonical source.
- [X] You are aged 18 or over.

**Comment by phazonoverload** [comment]
How will this extension handle auth?

**Comment by Theodore-Kelechukwu-Onyejiaku** [comment]
Hi @phazonoverload,

Thanks for asking.

So chrome extensions can store data in cookies or local storage. But for security, we will use cookies.

1. We will start with configuring Directus. We allowed CORS.
```
// configure directus
const client = createDirectus("http://0.0.0.0:8055")
  .with(authentication())
  .with(rest({ credentials: "include" }));
```
2. Make request to create a user. This will be in the registration page.
```
// create user
const result = await client.request(
  createUser({
    email: "hi@email.com",
    password: "qwerty123",
  }),
);
```
3. Then a user logs in to get their access token  and expiry time, which will be used to make authenticated requests.
```
// login user
const result = await client.login("admin@example.com", "d1r3ctu5");
const { access_token, expires_at } = result;
```

4. The access token and its expiry time will be saved to cookies using a third party cookie package.
```
// save token to cookie
Cookies.set("access_token", access_token, { expires: expires_at });
```

6. A User can then make requests using their access tokens.
```
// make request using token
const accessToken = Cookies.get("access_token");
const result = await client.request(
  withToken(accessToken, readItems("articles")),
);
```

7. When a user logs out, they no longer have the access token unless they log back in.
//  user logs out
```
Cookies.remove("access_token");
```

In summary, provided with the Directus server API, the extension can handle authentication using cookies and the Directus SDK.

Please let me know if this answers your question. Cheers.

**Comment by phazonoverload** [comment]
Oh yeah this is way more detail than I needed. Knowing you'll handle registration + login in the extension itself is plenty. 

Based on the agreed scope we can pay $350 for this post and the accompanying repository which you can create in your account and we will fork.

This post will be included as part of our May content schedule. The first draft is due on April 2 so we have time to properly review it and you have time to respond.

If interested and you’re happy to commit to the deadline, please let me know in the next few days and I will mark this issue as Approved.

The process is detailed in the README of this repo 😄

PS. Paying Nigerian bank accounts has continued to be a challenge as a non-Nigerian company. Having a US IBAN or a PayPal account seems to work best. Happy to discuss over email - kevin@directus.io

**Comment by phazonoverload** [comment]
Any update @Theodore-Kelechukwu-Onyejiaku?

**Comment by phazonoverload** [comment]
Given the lack of updates I'm going to assume you no longer have time for this. I'm going to close this issue out and remove it from our content schedule. If things change in the future and you wish to do this guest post, please let me know and I'll assess whether it works for us in terms of priorities, timeline, and budget.

**Comment by jaybharadia** [comment]
Hey @phazonoverload , Is this project still open for submission ? 
I would be happy to work in this project ( FYI: completed recent issue, submitted for review, meanwhile i am available for another work  )

**Comment by phazonoverload** [comment]
@jaybharadia not in the context of this submission, but feel free to open a new one with the same or similar idea and we will review it accordingly.
